### ISSUE TITLE
Remote references: Linter fails sporadically with empty file error

### ISSUE THREAD

**Issue by brendanmaguire** [zalando-incubator/restful-api-guidelines-linter#5]
Most of the time the linter work fine but every so often it fails with:

```
: Error during Swagger schema validation:
Expecting value: line 1 column 1 (char 0)Must: Provide API Reference Definition using OpenAPI
```

The file I am running it on is definitely not empty. For some reason it seems more likely to happen if the linter is run in quick succession.

**Comment by hjacobs** [comment]
@brendanmaguire maybe some other process (editor?) is rewriting the file while you run the linter? I can't reproduce the problem (running Ubuntu 16.04) with an endless loop:

``` bash
while true; do ./linter.py tests/fixtures/kio-api.yaml ; done
```

**Comment by brendanmaguire** [comment]
@hjacobs I've actually tracked it down to `.venv/lib/python3.5/site-packages/jsonschema/validators.py:421` . It reads https://zalando.github.io/problem/schema.yaml and tries to load it as json. The error I see above is just this exception, `json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)` , wrapped. I'm not sure why it's not always hitting this code path though as it succeeds sometimes for me.

**Comment by hjacobs** [comment]
@brendanmaguire ah, so you have a schema with a remote reference?

**Comment by brendanmaguire** [comment]
Yes, it references https://zalando.github.io/problem/schema.yaml#/Problem

**Comment by hjacobs** [comment]
OK, I could reproduce your problem as soon as I add a snippet like:

``` yaml
  responses:
      400:
          description: The foo body was invalid
          schema:
            $ref: 'https://zalando.github.io/problem/schema.yaml#/Problem'
```

**Comment by brendanmaguire** [comment]
Cool. That makes sense. The sporadic nature of the issue is strange, but at least it's reproducible.
